engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -312,30 +328,43 @@ class BindableTableScan : public kernel {
 		CodeTimer timer;
 
 		input.set_projections(get_projections(expression));
-		int batch_count = 0;
-		while (input.wait_for_next() ) {
-			try {
-				auto batch = input.next();
 
-				if(is_filtered_bindable_scan(expression)) {
-					auto columns = ral::processor::process_filter(batch->toBlazingTableView(), expression, context.get());
-					columns->setNames(fix_column_aliases(columns->names(), expression));
-					this->add_to_output_cache(std::move(columns));
-				}
-				else{
-					batch->setNames(fix_column_aliases(batch->names(), expression));
-					this->add_to_output_cache(std::move(batch));
+		int TABLE_SCAN_KERNEL_NUM_THREADS = 1;
+		std::map<std::string, std::string> config_options = context->getConfigOptions();
+		auto it = config_options.find("TABLE_SCAN_KERNEL_NUM_THREADS");
+		if (it != config_options.end()){
+			TABLE_SCAN_KERNEL_NUM_THREADS = std::stoi(config_options["TABLE_SCAN_KERNEL_NUM_THREADS"]);
+		}
+
+		std::vector<BlazingThread> threads;
+		for (int i = 0; i < TABLE_SCAN_KERNEL_NUM_THREADS; i++) {
+			threads.push_back(BlazingThread([expression = this->expression, this]() {
+				std::unique_ptr<ral::frame::BlazingTable> batch;
+				while(batch = input.next()) {
```

Comment:
Same as other scan, this is much less efficient than having 1 thread for most of the logic and seperate threads for the heavy lifting